### ISSUE TITLE
write_tables: avoid deadlock in case of errors [phab:D439]

### ISSUE THREAD

**Issue by kythe** [google/kythe#671]
_Created by **schroederc** at 2015-07-30 17:35:26_:

**Comment by kythe** [comment]
_**delroth** commented inline (context lost during migration) at 2015-07-30 18:06:32_:

FYI, that doesn't seem to build on go1.3.3. I had to replace it with for _ = range entries {}

**Comment by kythe** [comment]
_**delroth** wrote at 2015-07-30 18:06:32_:

```
2015/07/30 18:04:47 FATAL ERROR: error writing Edges: error writing rev edge: db writer close error: IO error: /tmp/reverse.edges989888102/001777.ldb: Too many open files
```

Seems to work. Thanks!

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-07-30 18:24:07_:

Use 1.3 syntax

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-07-30 18:24:21_:

Good point.  We're still targeting 1.3 for a while.

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2015-07-30 19:05:41_

**Comment by kythe** [comment]
_**schroederc** closed via commit 5c3078585e7e4faf10da0a2cebea4eff1b542cf2 at 2015-07-30 19:11:31_
